src/Viewer/components/MenuBar/MenuBar.js

```diff
@@ -135,6 +157,29 @@ export function MenuBar ({
         }
     };
 
+    const openCalendar = () => {
+        handleShowCalendar();
+    };
+
+    const closeCalendar = () => {
+        handleCloseCalendar();
+    };
+
+    const handleCalendarChange = (newDate) => {
+        if (newDate instanceof Date) {
+            setCalendarDateTime(newDate);
```

Comment:
use the millisecond truncation function here as well for consistent calendarDateTime state behavior (although maybe you already pass in a date value with the millisecond part removed).